### PR TITLE
Avogadro2: init at 1.94.0

### PR DESCRIPTION
**Pull request by sheepforce**
###### Motivation for this change
While Avogadro is in nixpkgs, the development was discontinued in favour of Avogadro2, which this PR adds. Contrary to Avogadro it builds using recent versions of OpenBabel and Eigen and offers some new functionalyties, while missing some that Avogadro1 has. Therefore both packages are still somewhat orthogonal.

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [21.11 Release Notes (or backporting 21.05 Relase notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2111-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by r-rmcgibbo** [comment]
Result of `nixpkgs-review pr 130359` at a992db24 run on aarch64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>5 packages built successfully:</summary>
  <ul>
    <li>avogadro2</li>
    <li>avogadrolibs</li>
    <li>libmsym</li>
    <li>mmtf-cpp</li>
    <li>molequeue</li>
  </ul>
</details>

File: pkgs/applications/science/chemistry/avogadro2/default.nix [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+    qt5.qttools
+  ];
+
+  propagatedBuildInputs = [ openbabel ];
+
+  postFixup = ''
+    wrapProgram $out/bin/avogadro2 \
+      --prefix PATH : "${openbabel}/bin"
+  '';
+
```
Comment (SuperSandro2000) [inline pkgs/applications/science/chemistry/avogadro2/default.nix:32]:
```suggestion
  qtWrapperArgs = [ "--prefix PATH : ${openbabel}/bin" ];
```

File: pkgs/applications/science/chemistry/avogadro2/default.nix [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+  version = "1.94.0";
+
+  src = fetchFromGitHub {
+    owner = "OpenChemistry";
+    repo = "avogadroapp";
+    rev = version;
+    sha256 = "6RaiX23YUMfTYAuSighcLGGlJtqeydNgi3PWGF77Jp8=";
+  };
+
+  nativeBuildInputs = [ cmake wrapQtAppsHook makeWrapper ];
```
Comment (SuperSandro2000) [inline pkgs/applications/science/chemistry/avogadro2/default.nix:16]:
```suggestion
  nativeBuildInputs = [ cmake wrapQtAppsHook ];
```

File: pkgs/development/libraries/science/chemistry/libmsym/default.nix [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+    owner = "mcodev31";
+    repo = pname;
+    rev = "v${version}";
+    sha256= "k+OEwrA/saupP/wX6Ii5My0vffiJ0X9xMCTrliMSMik=";
+  };
+
+  nativeBuildInputs = [ cmake ];
+
+  buildInputs = [ ];
+
```
Comment (SuperSandro2000) [inline pkgs/development/libraries/science/chemistry/libmsym/default.nix:17]:
```suggestion
```

**Review by SuperSandro2000 — CHANGES_REQUESTED** [review]

File: pkgs/development/libraries/science/chemistry/avogadrolibs/default.nix [hunk 4]
```diff
@@ -0,0 +1,67 @@
+{ lib, stdenv, fetchFromGitHub, cmake, zlib, eigen, libGL, doxygen, spglib
+, mmtf-cpp, glew, python3, libarchive, libmsym, msgpack, qt5, wrapQtAppsHook
+}:
+
+let
+  pythonWP = python3.withPackages (p: with p; [ openbabel-bindings numpy ]);
```
Comment (Artturin) [inline pkgs/development/libraries/science/chemistry/avogadrolibs/default.nix:6]: this doesn't seem to be used anywhere

Comment (sheepforce) [reply pkgs/development/libraries/science/chemistry/avogadrolibs/default.nix]: You are right. Was introduced when I've searched for the OpenBabel bindings. Has been removed.

File: pkgs/development/libraries/science/chemistry/molequeue/default.nix [hunk 5]
```diff
@@ -0,0 +1,32 @@
...
+
+  buildInputs = [ qt5.qttools ];
+
+  postFixup = ''
+    substituteInPlace $out/lib/cmake/molequeue/MoleQueueConfig.cmake \
+      --replace "''${MoleQueue_INSTALL_PREFIX}/$out" "''${MoleQueue_INSTALL_PREFIX}"
+  '';
+
+  meta = with lib; {
+    description = "Desktop integration of high performance computing resources";
```
Comment (SuperSandro2000) [inline pkgs/development/libraries/science/chemistry/molequeue/default.nix:27]: homepage is missing?

File: pkgs/development/libraries/science/chemistry/molequeue/default.nix [hunk 6]
```diff
@@ -0,0 +1,32 @@
...
+    rev = version;
+    sha256 = "+NoY8YVseFyBbxc3ttFWiQuHQyy1GN8zvV1jGFjmvLg=";
+  };
+
+  nativeBuildInputs = [
+    cmake
+    wrapQtAppsHook
+  ];
+
+  buildInputs = [ qt5.qttools ];
```
Comment (SuperSandro2000) [inline pkgs/development/libraries/science/chemistry/molequeue/default.nix:19]:
```suggestion
  buildInputs = [ qttools ];
```

File: pkgs/development/libraries/science/chemistry/molequeue/default.nix [hunk 7]
```diff
@@ -0,0 +1,32 @@
+{ lib, stdenv, fetchFromGitHub, cmake, qt5, wrapQtAppsHook }:
```
Comment (SuperSandro2000) [inline pkgs/development/libraries/science/chemistry/molequeue/default.nix:1]:
```suggestion
{ lib, stdenv, fetchFromGitHub, cmake, qttools, wrapQtAppsHook }:
```

File: pkgs/development/libraries/science/chemistry/avogadrolibs/default.nix [hunk 8]
```diff
@@ -0,0 +1,65 @@
...
+    qt5.qttools
+  ];
+
+  postFixup = ''
+    substituteInPlace $out/lib/cmake/${pname}/AvogadroLibsConfig.cmake \
+      --replace "''${AvogadroLibs_INSTALL_PREFIX}/$out" "''${AvogadroLibs_INSTALL_PREFIX}"
+  '';
+
+  meta = with lib; {
+    description = "Molecule editor and visualizer";
```
Comment (SuperSandro2000) [inline pkgs/development/libraries/science/chemistry/avogadrolibs/default.nix:60]: Missing homepage.

File: pkgs/development/libraries/science/chemistry/avogadrolibs/default.nix [hunk 9]
```diff
@@ -0,0 +1,65 @@
...
+    eigen
+    zlib
+    libGL
+    spglib
+    mmtf-cpp
+    glew
+    libarchive
+    libmsym
+    msgpack
+    qt5.qttools
```
Comment (SuperSandro2000) [inline pkgs/development/libraries/science/chemistry/avogadrolibs/default.nix:51]:
```suggestion
    qttools
```

File: pkgs/development/libraries/science/chemistry/avogadrolibs/default.nix [hunk 10]
```diff
@@ -0,0 +1,65 @@
+{ lib, stdenv, fetchFromGitHub, cmake, zlib, eigen, libGL, doxygen, spglib
+, mmtf-cpp, glew, libarchive, libmsym, msgpack, qt5, wrapQtAppsHook
```
Comment (SuperSandro2000) [inline pkgs/development/libraries/science/chemistry/avogadrolibs/default.nix:2]:
```suggestion
, mmtf-cpp, glew, libarchive, libmsym, msgpack, qttools, wrapQtAppsHook
```

File: pkgs/applications/science/chemistry/avogadro2/default.nix [hunk 11]
```diff
@@ -0,0 +1,36 @@
...
+  };
+
+  nativeBuildInputs = [ cmake wrapQtAppsHook ];
+
+  buildInputs = [
+    avogadrolibs
+    molequeue
+    eigen
+    hdf5
+    qt5.qttools
```
Comment (SuperSandro2000) [inline pkgs/applications/science/chemistry/avogadro2/default.nix:23]:
```suggestion
    qttools
```

File: pkgs/applications/science/chemistry/avogadro2/default.nix [hunk 12]
```diff
@@ -0,0 +1,36 @@
+{ lib, stdenv, fetchFromGitHub, cmake, eigen, avogadrolibs, molequeue, hdf5
+, openbabel, qt5, wrapQtAppsHook
```
Comment (SuperSandro2000) [inline pkgs/applications/science/chemistry/avogadro2/default.nix:2]:
```suggestion
, openbabel, qttools, wrapQtAppsHook
```

**Review by SuperSandro2000 — CHANGES_REQUESTED** [review]

**Review by markuskowa — APPROVED** [review]
Looks good to me.

**Pull request merged**